### PR TITLE
BUG: fix typo in polydiv that prevented promotion to poly1d

### LINKED ISSUE #17076: BUG: Typo in `numpy.polydiv`
https://github.com/numpy/numpy/blob/92ebe1e9a6aeb47a881a1226b08218175776f9ea/numpy/lib/polynomial.py#L959

The typo in this line causes wrong return types in cases when `v` is `poly1d`

fix to be: `truepoly = (isinstance(u, poly1d) or isinstance(v, poly1d))`

### PR DESCRIPTION
**Pull request by jakobjakobson13**
See  #17012 item 16.

Closes #17076

### CODE REVIEW

File: numpy/lib/polynomial.py [hunk 1]
```diff
@@ -1017,7 +1017,7 @@ def polydiv(u, v):
     (array([1.5 , 1.75]), array([0.25]))
 
     """
-    truepoly = (isinstance(u, poly1d) or isinstance(u, poly1d))
```
Comment (eric-wieser) [inline numpy/lib/polynomial.py:1020]: I think this was a typo and the second `u` should be `v`

Comment (jakobjakobson13) [reply numpy/lib/polynomial.py:1020]: This would indeed change the comparison. Shall it change it to `(isinstance(u, poly1d) or isinstance(v, poly1d))` ?

Comment (eric-wieser) [reply numpy/lib/polynomial.py:1020]: Yes please.

**Comment by eric-wieser** [comment]
It might be good to add a test for the return type of `polydiv(some_arr, some_poly)`, which this patch changes.

**Comment by Dahlia-Chehata** [comment]
closes #17076

**Comment by eric-wieser** [comment]
@Dahlia-Chehata: Not sure it was really worth opening an issue about that given we already have the patch, but no harm done.

**Comment by Dahlia-Chehata** [comment]
@eric-wieser you are right, but I discovered this PR after I already had opened the issue.

File: numpy/lib/tests/test_polynomial.py [hunk 2]
```diff
@@ -243,6 +243,12 @@ def test_polydiv(self):
         assert_equal(q.coeffs.dtype, np.complex128)
         assert_equal(r.coeffs.dtype, np.complex128)
         assert_equal(q*a + r, b)
+        
+        d = np.poly1d([1, 2, 3])
+        c = [1, 2, 3]
+        s, t = np.polydiv(d, c)
+        assert_equal(isinstance(s, np.poly1d) == True)
+        assert_equal(isinstance(t, np.poly1d) == True)
```
Comment (eric-wieser) [inline numpy/lib/tests/test_polynomial.py:251]:
```suggestion
        assert isinstance(s, np.poly1d)
        assert isinstance(t, np.poly1d)
```

File: numpy/lib/tests/test_polynomial.py [hunk 3]
```diff
@@ -243,6 +243,12 @@ def test_polydiv(self):
         assert_equal(q.coeffs.dtype, np.complex128)
         assert_equal(r.coeffs.dtype, np.complex128)
         assert_equal(q*a + r, b)
+        
+        d = np.poly1d([1, 2, 3])
+        c = [1, 2, 3]
+        s, t = np.polydiv(d, c)
```
Comment (eric-wieser) [inline numpy/lib/tests/test_polynomial.py:249]: This test is backwards - its the one that passed before this patch.

Probably fine to keep it, but you should add one with the arguments in the other order too.

**Review by rossbar — APPROVED** [review]
LGTM, thanks @jakobjakobson13

**Pull request merged**